### DISCUSSION TITLE
Is milvus comparable with FAISS by performance when searching by itself in FLAT index. The task to search all the vector against the whole base to find very close neighbors.

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#34571] [Q&A and General discussion]
Is milvus comparable with FAISS by performance when searching by itself in FLAT index. The task to search all the vector against the whole base to find very close neighbors. 

So I have like  10 millions  of 512 size vector.
And want to search against each of them.

so essentially it 10 million operations. Each takes around 1sec to complete.

Is milvus will be much slower for this task or will it be comparable to FAISS? 
I mean the network overhead is negligible  in my case and could be ignored?

If FAISS (or some local alternative) is much faster what is the best way to export all vectors from Milvus?

**Comment by yhmo** [comment]
Milvus manages data by segments, search request is executed in each segment, returns N topk results, and the N results are combined into a final topk. And milvus is client/server, there are lot of overhead in rpc transfer and system management.
FAISS can treat the entire dataset as a large in-memory block if you have sufficient memory. It is much faster than milvus for brute-force search.

The best way to export all vectors from milvus is query iterator: https://milvus.io/docs/with-iterators.md#Query-with-an-iterator

**Comment by xiaofan-luan** [comment]
1. I thought for 10M data, milvus has comparable performance than faiss if both using FLAT.
2. Export and flat search is two different things, as @yhmo said, iterator is what you need to export all data.
